sentry/src/main/java/io/sentry/Attachment.java

```diff
@@ -0,0 +1,109 @@
+package io.sentry;
+
+import java.io.File;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** You can use an attachment to store additional files alongside an event or transaction. */
+@ApiStatus.Experimental
+public final class Attachment {
+
+  private @Nullable byte[] bytes;
+  private @Nullable String path;
+  private final String filename;
+  private String contentType;
+
+  /**
+   * We could use Files.probeContentType(path) to determine the content type of the filename. This
+   * needs a path, but file.toPath or Paths.get only work on above Android API level 26, see
+   * https://developer.android.com/reference/java/nio/file/Paths. There are also ways via
+   * URLConnection, but we don't want to use this in constructors. Therefore we use the default
+   * content type of Sentry.
+   */
+  private static final String DEFAULT_CONTENT_TYPE = "application/octet-stream";
+
+  /**
+   * Initializes an Attachment with bytes.
+   *
+   * <p>The file located at the path is read lazily when the SDK captures an event or transaction
+   * not when the attachment is initialized.
+   *
+   * @param bytes The bytes of file.
+   * @param filename The name of the attachment to display in Sentry.
+   */
+  public Attachment(@NotNull byte[] bytes, @NotNull String filename) {
```

Comment:
```suggestion
  public Attachment(final @NotNull byte[] bytes, final @NotNull String filename) {
```

and any other missing param